src/explore-education-statistics-admin/src/pages/release/content/components/__tests__/EditableKeyStatText.test.tsx

```diff
@@ -167,4 +171,160 @@ describe('EditableKeyStatText', () => {
       screen.queryByRole('button', { name: /Remove/ }),
     ).not.toBeInTheDocument();
   });
+
+  describe('when editing', () => {
+    test('renders edit form correctly', async () => {
+      render(
+        <EditableKeyStatText keyStat={keyStatText} isEditing onSubmit={noop} />,
+      );
+
+      await waitFor(() => {
+        expect(screen.getByTestId('keyStat-title')).toHaveTextContent(
+          'Text title',
+        );
+        expect(screen.getByTestId('keyStat-statistic')).toHaveTextContent(
+          'Over 9000',
+        );
+        expect(screen.getByTestId('keyStat-trend')).toHaveTextContent(
+          'Text trend',
+        );
+        expect(
+          screen.getByRole('button', {
+            name: 'Text guidance title',
+          }),
+        ).toBeInTheDocument();
+
+        expect(screen.getByTestId('keyStat-guidanceText')).toHaveTextContent(
+          'Text guidance text',
+        );
+
+        expect(
+          screen.getByRole('button', {
+            name: 'Edit key statistic: Text title',
+          }),
+        ).toBeInTheDocument();
+
+        expect(
+          screen.queryByRole('button', { name: /Remove/ }),
+        ).not.toBeInTheDocument();
+      });
+    });
+
+    test('submitting edit form calls `onSubmit` handler with updated values', async () => {
+      const onSubmit = jest.fn();
+
+      render(
+        <EditableKeyStatText
+          keyStat={keyStatText}
+          isEditing
+          onSubmit={onSubmit}
+        />,
+      );
+
+      await waitFor(() => {
+        expect(screen.getByText(/Edit/)).toBeInTheDocument();
+      });
+
+      userEvent.click(
+        screen.getByRole('button', {
+          name: 'Edit key statistic: Text title',
+        }),
+      );
+
+      await waitFor(() => {
+        expect(screen.getByLabelText('Title')).toBeInTheDocument();
+      });
+
+      await userEvent.clear(screen.getByLabelText('Title'));
```

Comment:
Don't know if we need to await all these userEvents

Reply:
Did you see if these awaits are all necessary? VSCode reckons the ones on userEvent.clear aren't:
![await](https://user-images.githubusercontent.com/81572860/220980771-bdb1657b-b332-46a6-b2a1-7f9be0957e86.PNG)

Reply:
Seems VSCode is right - I'll remove them from EditableKeyStatDataBlock.test.tsx too.